scripts/cleanup-workload-identity.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Install kubectl and kind
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# shellcheck source=hack/ensure-azcli.sh
+source "${REPO_ROOT}/hack/ensure-azcli.sh"
+
+AZWI_RESOURCE_GROUP="${AZWI_RESOURCE_GROUP:-}"
+
+if [[ -z "${AZWI_RESOURCE_GROUP}" ]]; then
+  echo AZWI_RESOURCE_GROUP environment variable must be set
+  exit 1
```

Comment:
To clarify, if I want to bring my own WI setup then I don't need to set `AZWI_RESOURCE_GROUP`? Should we make this a clean exit 0 if that's true?

Reply:
I think it's fair to require this var in this script. I'll add a SKIP_CLEANUP flag before we call this script for BYO scenarios.

Reply:
Or maybe `AZWI_SKIP_CLEANUP` since `SKIP_CLEANUP` already means something else. I can see if I run an e2e  test locally with my pre-existing workload ID setup where I'd want to test deleting the Clusters created by the test but I still wouldn't want to delete my workload ID setup.

Reply:
I like that idea